sql/core/src/main/scala/org/apache/spark/sql/execution/SparkOptimizer.scala

```diff
@@ -50,7 +50,8 @@ class SparkOptimizer(
   override def defaultBatches: Seq[Batch] = (preOptimizationBatches ++ super.defaultBatches :+
     Batch("Optimize Metadata Only Query", Once, OptimizeMetadataOnlyQuery(catalog)) :+
     Batch("PartitionPruning", Once,
-      PartitionPruning) :+
+      PartitionPruning,
+      RowLevelOperationRuntimeGroupFiltering(OptimizeSubqueries)) :+
```

Comment:
I think another idea is to run `OptimizeSubqueries` in this batch:
```
PartitionPruning,
RowLevelOperationRuntimeGroupFiltering,
// The rule above may create subqueries, need to optimize them.
OptimizeSubqueries
```

Reply:
This would be much cleaner but SPARK-36444 removed `OptimizeSubqueries`. I can add a separate batch for group filtering in row-level operations. How does that sound, @cloud-fan?